Magento_Theme/web/js/view/custom-messages.js

```diff
@@ -0,0 +1,52 @@
+define([
+  'jquery',
+  'uiComponent',
```

Comment:
you duplicated the whole component but you also can extend the existing custom one, like
```js
define(['Magento_Ui/js/view/messages'], (Messages) => {
 return Messages.extend({ ... });
}
```
via this trick you don't have to duplicate everything, only override what you need.
also you will be available to use `this._super();` for methods from the original message.js script to call the original logic for that methods what you want to override.